app.py

```
@@ -168,78 +172,191 @@ def get_config():
-    })
-
 @app.route('/api/profile')
 def api_profile():
     """API endpoint to get the current user's profile info."""
     user_emoji = session.get('logged_in_user_emoji')
     if not user_emoji:
         return jsonify({"error": "Not authenticated"}), 401
 
+    try:
+        user_res = supabase.table('user').select('*').eq('emoji', user_emoji).single().execute()
+        if not user_res.data: return jsonify({"error": "User not found"}), 404
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)